lib/trivial.nix

```diff
@@ -1,17 +1,44 @@
 rec {
 
-  # Identity function.
+  /* The identity function
+     For when you need a function that does “nothing”.
+
+     Type: id :: a -> a
+  */
   id = x: x;
 
-  # Constant function.
+  /* The constant function
+     Ignores the second argument.
+     Or: Construct a function that always returns a static value.
+
+     Type: const :: a -> b -> a
```

Comment:
At least in Haskell, where this style of type signatures comes from. In nix it’s possible of course (with type introspection). But that’s evil. :)